class.jetpack.php

```diff
@@ -4790,23 +4790,13 @@ public static function api_url( $relative_url ) {
 	}
 
 	/**
+	 * @deprecated 7.8 Use Automattic\Jetpack\Connection\Manager::fix_url_for_bad_hosts() instead.
+     *
 	 * Some hosts disable the OpenSSL extension and so cannot make outgoing HTTPS requsets
 	 */
 	public static function fix_url_for_bad_hosts( $url ) {
-		if ( 0 !== strpos( $url, 'https://' ) ) {
-			return $url;
-		}
-
-		switch ( JETPACK_CLIENT__HTTPS ) {
-			case 'ALWAYS' :
-				return $url;
-			case 'NEVER' :
-				return set_url_scheme( $url, 'http' );
-			// default : case 'AUTO' :
-		}
-
-		// we now return the unmodified SSL URL by default, as a security precaution
-		return $url;
+		_deprecated_function( __METHOD__, 'jetpack-7.8', 'Automattic\Jetpack\Connection\Manager' );
```

Comment:
Let's make sure to include the method too in the third argument, see other deprecated method examples in this same file.

Also, let's be careful about backslashes in single quotes; in order to represent a real backslash, you need to specify them twice in that context:

```suggestion
		_deprecated_function( __METHOD__, 'jetpack-7.8', 'Automattic\\Jetpack\\Connection\\Manager::fix_url_for_bad_hosts' );
```

Reply:
See 57bcd35.